torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -264,6 +264,36 @@ std::shared_ptr<SugaredValue> ClassValue::attr(
   }
   return std::make_shared<ClassNewMethod>(type_);
 }
+
+std::shared_ptr<SugaredValue> NamedTupleValue::call(
+    const SourceRange& loc,
+    Function& m,
+    at::ArrayRef<NamedValue> inputs,
+    at::ArrayRef<NamedValue> attributes,
+    size_t n_binders) {
+  auto nargs = type_->containedTypes().size();
+  if (inputs.size() != nargs) {
+    throw ErrorReport(loc) << "Constructor expected " << nargs << "arguments "
+                           << "but got " << inputs.size();
+  }
+  AT_CHECK(attributes.size() == 0);
```

Comment:
`MyNamedTuple(a=3) ` is going to error without line information.
This patch: make this an error report. In the future, we should be supporting keyword construction of tuples.